src/routes/transactions/mappers/multisig-transaction.mapper.ts

```diff
@@ -0,0 +1,592 @@
+          type: 'SET_FALLBACK_HANDLER',
+          handler: dataDecoded.parameters[0]?.value,
+        };
+      case this.ADD_OWNER_WITH_THRESHOLD:
+        return <AddOwner>{
+          type: 'ADD_OWNER',
+          owner: { value: dataDecoded.parameters[0]?.value },
+          threshold: Number(dataDecoded.parameters[1]?.value),
+        };
+      case this.REMOVE_OWNER:
+        return <RemoveOwner>{
+          type: 'REMOVE_OWNER',
+          owner: { value: dataDecoded.parameters[1]?.value },
+          threshold: Number(dataDecoded.parameters[2]?.value),
+        };
+      case this.SWAP_OWNER:
+        return <SwapOwner>{
+          type: 'SWAP_OWNER',
+          oldOwner: { value: dataDecoded.parameters[1]?.value },
+          newOwner: { value: dataDecoded.parameters[2]?.value },
+        };
+      case this.CHANGE_MASTER_COPY:
+        return <ChangeImplementation>{
+          type: 'CHANGE_MASTER_COPY',
+          implementation: await this.addressInfoHelper.getOrDefault(
+            chainId,
+            safe.address,
+          ),
+        };
+      case this.ENABLE_MODULE:
+        return <EnableModule>{
+          type: 'ENABLE_MODULE',
+          module: await this.addressInfoHelper.getOrDefault(
+            chainId,
+            dataDecoded.parameters[0]?.value,
+          ),
+        };
+      case this.DISABLE_MODULE:
+        return <DisableModule>{
+          type: 'DISABLE_MODULE',
+          module: await this.addressInfoHelper.getOrDefault(
+            chainId,
+            dataDecoded.parameters[1]?.value,
+          ),
+        };
+      case this.CHANGE_THRESHOLD:
+        return <ChangeThreshold>{
+          type: 'CHANGE_THRESHOLD',
+          threshold: dataDecoded.parameters[0]?.value,
+        };
+      case this.SET_GUARD:
+        const guardValue = dataDecoded.parameters[0]?.value;
+        if (guardValue !== this.NULL_ADDRESS) {
+          return <SetGuard>{
+            type: 'SET_GUARD',
+            guard: await this.addressInfoHelper.getOrDefault(
+              chainId,
+              guardValue,
+            ),
+          };
+        }
+        return <DeleteGuard>{
+          type: 'DELETE_GUARD',
+        };
+    }
+  }
+
+  private mapTxStatus(transaction: MultisigTransaction, safe: Safe): string {
+    if (transaction.isExecuted) {
+      return transaction.isSuccessful ? 'SUCCESS' : 'FAILED';
+    }
+    if (safe.nonce > transaction.nonce) {
+      return 'CANCELLED';
+    }
+    if (
+      this.getConfirmationsCount(transaction) <
+      this.getConfirmationsRequired(transaction, safe)
+    ) {
+      return 'AWAITING_CONFIRMATIONS';
+    }
+    return 'AWAITING_EXECUTION';
+  }
+
+  private getConfirmationsCount(transaction: MultisigTransaction): number {
+    return transaction.confirmations?.length || 0;
+  }
+
+  private getConfirmationsRequired(
+    transaction: MultisigTransaction,
+    safe: Safe,
+  ): number {
+    return transaction.confirmationsRequired ?? safe.threshold;
```

Comment:
Good question. Maybe when the transaction was created the number of required signers is recorded in the `confirmationsRequired` field, and It could be different to `safe.threshold` because the Safe configuration could change after the transaction was created. Is that possible?

Reply:
With caching in place for both endpoints it's hard to say. I'd say to leave a `TODO` here so we can come back to this later.

But It'd be better to rely only on one source.

Reply:
Added a TODO in [4827c02](https://github.com/5afe/safe-client-gateway-nest/pull/197/commits/4827c02120e9672707846c48d0552b30dd1952cb)